File: MD.CMS.BusinessLogic.Core/DataAccess/Entities/ReportDefinition.cs
using MD.Tools.BaseDataAccess.Core.Entities;
using MD.Tools.BaseDataAccess.Plugins.Core.Mapping.ReportDefinition.ReportDesigner;
using Newtonsoft.Json;
using System;

namespace MD.CMS.BusinessLogic.Core.DataAccess.Entities
{
    public class ReportDefinition : BaseEntity<long>
    {
        #region Attributes
        private string _name;
        private InnerReportDefinition _definition;
        private string _sql;
        private string _authorId;
        private User _author;
        private DateTime _dateCreated;
        private DateTime _dateModified;
        #endregion

        #region Properties
        public string Name
        {
            get { return _name; }
            set { _name = value; }
        }
        public InnerReportDefinition Definition
        {
            get { return _definition; }
            set { _definition = value; }
        }
        
        public string Sql
        {
            get { return _sql; }
            set { _sql = value; }
        }
        
        public string Json
        {
            get 
            {
                if (_definition != null)
                {
                    return JsonConvert.SerializeObject(_definition);
                }
                return string.Empty;
            }
            set 
            {
                try
                {
                    _definition = JsonConvert.DeserializeObject<InnerReportDefinition>(value);
                }
                catch (Exception error)
                {

                }
            }
        }

        public string AuthorId
        {
            get { return _authorId; }
            set { _authorId = value; }
        }

        public User Author
        {
            get { return _author; }
            set { _author = value; }
        }

        public DateTime DateCreated
        {
            get { return _dateCreated; }
            set { _dateCreated = value; }
        }

        public DateTime DateModified
        {
            get { return _dateModified; }
            set { _dateModified = value; }
        }
        #endregion
    }
}

File: MD.CMS.Administration/MD.CMS.Administration.Core.Web/scripts/businessLogic/dataAccess/entities/reportDefinition.ts
/// <reference path="./base/iBaseEntity.ts" />
/// <reference path="./base/baseEntity.ts" />
/// <reference path="./user.ts" />
/// <reference path="./innerReportDefinition.ts" />

namespace mdBusinessLogic.dataAccess.entities {
    export class reportDefinition extends base.BaseEntity implements base.IBaseEntity<reportDefinition> {
        public Name: string;
        public Definition: innerReportDefinition;
        public Sql: string;
        public AuthorId: number;
        public Author: user;
        public Json: string;
        public DateCreated: Date;
        public DateModified: Date;

        constructor(obj?: reportDefinition) {
            super(obj);
            this.Name = '';
            this.Definition = new innerReportDefinition();
            this.Sql = '';
            this.AuthorId = 0;
            this.Author = new user();
            this.Json = '';
            this.DateCreated = new Date();
            this.DateModified = new Date();
            if (obj != undefined && obj != null) {
                this.construct(obj);
            }
        }

        construct(data: any): void {
            super.construct(data);
            this.Name = this.getValue<string>(data, "Name", '');
            this.Definition = this.getConstructEntityValue<innerReportDefinition>(data, "Definition", new innerReportDefinition());
            this.Sql = this.getValue<string>(data, "Sql", '');
            this.AuthorId = this.getValue<number>(data, "AuthorId", 0);
            this.Author = this.getConstructEntityValue<user>(data, "Author", new user());
            this.Json = this.getValue<string>(data, "Json", '');
            this.DateCreated = this.getValue<Date>(data, "DateCreated", new Date());
            this.DateModified = this.getValue<Date>(data, "DateModified", new Date());
        }

        public clone(): reportDefinition {
            return new reportDefinition(this);
        }

    }
}